cpp/src/com/google/idea/blaze/cpp/BlazeCWorkspace.java

```diff
@@ -165,6 +172,97 @@ public void run(ProgressIndicator indicator) {
             });
   }
 
+  private Path trimStart(Path value, @Nullable Path prefix) {
+    if (prefix == null || !value.startsWith(prefix)) {
+        return value;
+    }
+
+    return value.subpath(prefix.getNameCount(), value.getNameCount());
+  }
+
+  private @Nullable Path pathOf(@Nullable String value) {
+    if (value == null || value.isEmpty()) {
+        return null;
+    }
+
+    try {
+      return Path.of(value);
+    } catch (InvalidPathException e) {
+      return null;
+    }
+  }
+
+  private List<String> collectIncludes(Path root, TargetKey targetKey, BlazeProjectData blazeProjectData) {
```

Comment:
I'd pass indicator to this routine as well and performed a check `indicator.checkCanceled()` in both `for` loops.